docs/updating/nim-migration/custom-plugins.md

```diff
@@ -9,4 +9,141 @@ modified_at: 2024-06-06
 
 # Migrating custom plugins
 
-Placeholder
+If you have created custom plugins for CKEditor&nbsp;5, you will need to adjust them to make them work with the new installation methods. The migration process is slightly different depending on whether you only want to support the new installation methods or also maintain backwards compatibility with the old installation methods.
+
+<info-box warning>
+	This migration guide assumes that you have created a custom plugin using our {@link framework/development-tools/package-generator/using-package-generator package generator}. If you created your plugin in a different way, you will need to adjust the steps accordingly.
+</info-box>
+
+## Prerequisites
+
+Before you start, follow the usual upgrade path to update your plugin to use the latest version of CKEditor&nbsp;5. This will rule out any problems that may be caused by upgrading from an outdated version of CKEditor&nbsp;5.
+
+## Migration steps
+
+### Create a new project using package generator
+
+To ensure that all the dependencies are up-to-date and that the build process is correct, we recommend that you:
+
+1. Create a new project using the package generator following the {@link framework/development-tools/package-generator/using-package-generator package generator guide}.
+2. Copy the `src`, `tests`, and `sample` folders of your plugin into the new project.
+3. Re-add all the external `dependencies`, `devDependencies`, and `peerDependencies` specific to your plugin to the `package.json` file.
+
+When you run the CLI, you will be asked to choose whether you want to support only the new installation methods or provide backward compatibility with the old installation methods. Choose the option that best suits your needs, but be aware that the latter option will generate additional files and code that you will need to update or remove later. However, you should consider this if your plugin is used in projects outside your control that may still use the old installation methods, for example if your plugin is open-source.
+
+The main changes we have introduced in the new package generator are:
+
+* Making the generated package a valid ECMAScript module,
+* Updating the build process to generate bundles for the new installation methods,
+* Adding new eslint rules to avoid common errors,
+* Updating dependencies.
+
+### Add missing file extensions in imports
+
+Next, add the missing file extensions in imports to all files in the `src`, `tests`, and `sample` folders. 
```

Comment:
Please add at least one why sentence. Why I need to do this step.

Reply:
Ok, I see it's mentioned later... Hmm, maybe add `...folders to make it a proper ESM import.`

Reply:
Done :heavy_check_mark: